plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceRuntimeInfo.java

```diff
@@ -310,16 +316,33 @@ public String projectsRoot() {
     protected Map<String, ServerImpl> getServersWithFilledPorts(final String externalHostame, final String internalHostname, final Map<String, List<PortBinding>> exposedPorts) {
         final HashMap<String, ServerImpl> servers = new LinkedHashMap<>();
 
+        boolean useMappedPorts = true;
+        if (useInternalContainerAddresses()) {
+            if (info.getNetworkSettings() != null && internalHostname.equals(info.getNetworkSettings().getIpAddress())) {
+                useMappedPorts = false;
```

Comment:
This was meant mostly as a sanity check, although even as that I'm not sure it's handled as well as it could be. `internalHostname` is still the address used to communicate with wsagent, and I was concerned that it may be possible to create an instance of `DockerInstanceRuntimeInfo` with `internalHostname` not matching `info.getNetworkSettings().getIpAddress()`, in which case contacting wsagent would fail even if `internalHostname` is valid. 

If you feel that it's not a concern, I will remove it.